lib/Controller/WopiController.php

```diff
@@ -485,10 +485,21 @@ public function putFile($fileId,
 
 			$content = fopen('php://input', 'rb');
 
+			// Get expected FileSize after writing
+			$storage = \OC_Helper::getStorageInfo($path);
+			fseek($content,0,SEEK_END);
+			$finalSize = ftell($content) + $file->getSize();
+			rewind( $content );
```

Comment:
I'm not fully sure but remember something around the input stream being only readable once, however I could only find a reference to this prior to PHP 5.6.

https://github.com/php/doc-en/commit/cdaea0421544885f02ff3d36bd203dc01b78299e#diff-f476ea6ab31592691b4878cd08dd817003bef1fa186317914fc0e94d51bca52fL56-L62

@icewind1991 Do you happen to know if this works fine?

Reply:
@icewind1991 so we should find another way to read out the size, right? is there any better way to do it? Maybe read out the stream once and than pass it on as a string instead of a stream?

Reply:
Passing into a string will potentially cause issues if the file size exceeds the memory limit of the php process.

We could write it to a temporary file and use the size of that but that has some overhead in copying and writing/reading that file of course.

As an alternative we could check if Collabora actually sends a Content-Length header that we can just use, otherwise we may want to copy to a temporary stream and use that.

Reply:
CONTENT_LENGTH header is indeed set, changed it to use that.